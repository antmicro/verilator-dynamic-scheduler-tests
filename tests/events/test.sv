module t (/*AUTOARG*/
   // Inputs
   clk
   );
   input clk;
   event e;

   initial begin
     #1 $write("Will trigger the event in #100\n");
     #100 $write("Triggering!\n");
     ->e;
     $write("Will wait for the event here...\n");
     @e $write("Got the event!\n");
     $write("*-* All Finished *-*\n");
     $finish;
   end

   initial begin
     $write("Will wait for the event here...\n");
     @e;
     $write("Got the event!\n");
     #1 $write("Will trigger the event in #100\n");
     #100 $write("Triggering!\n");
     ->e;
   end
endmodule
