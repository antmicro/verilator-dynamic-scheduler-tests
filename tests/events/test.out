Will wait for the event here...
Will trigger the event in #100
Triggering!
Got the event!
Will wait for the event here...
Will trigger the event in #100
Triggering!
Got the event!
*-* All Finished *-*
